hyperspy/io_plugins/semper_unf.py

```diff
@@ -0,0 +1,563 @@
+                    ('Z0V4', ('<i2', 4)),     # Bytes 80-83
+                    ('DYV3', ('<i2', 4)),     # Bytes 84-87
+                    ('Y0V2', ('<i2', 4)),     # Bytes 88-91
+                    ('DXV1', ('<i2', 4)),     # Bytes 92-95
+                    ('X0V0', ('<i2', 4)),     # Bytes 96-99
+                    ('NTITLE', '<i2'),        # Bytes 100
+                    ('TITLE', ('<i2', 144)),  # Bytes 101-244
+                    ('XUNIT', ('<i2', 4)),    # Bytes 245-248
+                    ('YUNIT', ('<i2', 4)),    # Bytes 249-252
+                    ('ZUNIT', ('<i2', 4))]    # Bytes 253-256
+
+    def __init__(self, data, title=Undefined, offsets=(0., 0., 0.), scales=(1., 1., 1.),
+                 units=(Undefined, Undefined, Undefined), metadata=None):
+        self._log.debug('Calling __init__')
+        if metadata is None:
+            metadata = {}
+        data = data[tuple(None for _ in xrange(3-len(data.shape)))]  # Make sure data is 3D!
+        self.data = data
+        self.title = title
+        self.offsets = offsets
+        self.scales = scales
+        self.units = units
+        self.metadata = metadata
+        self._log.debug('Created '+str(self))
+
+    @classmethod
+    def _read_label(cls, unf_file):
+        rec_length = np.fromfile(unf_file, dtype='<i4', count=1)[0]  # length of label
+        label = sarray2dict(np.fromfile(unf_file, dtype=cls.LABEL_DTYPES, count=1))
+        label.update({'SEMPER': ''.join([str(unichr(l)) for l in label['SEMPER']])})
+        assert label['SEMPER'] == 'Semper'
+        # Process dimensions:
+        for key in ['NCOL', 'NROW', 'NLAY', 'ICCOLN', 'ICROWN', 'ICLAYN']:
+            value = 256**2*label.pop(key+'H') + 256*label[key][0] + label[key][1]
+            label.update({key: value})
+        # Process date:
+        date = '{}-{}-{} {}:{}:{}'.format(label['DATE'][0]+1900, *label['DATE'][1:])
+        label.update({'DATE': date})
+        # Process range:
+        if label['NCRANG'] == 255:
+            range_min = struct.unpack('<f', ''.join([chr(x) for x in label['RANGE'][:4]]))[0]
+            range_max = struct.unpack('<f', ''.join([chr(x) for x in label['RANGE'][4:8]]))[0]
+            range_string = '{:.6g},{:.6g}'.format(range_min, range_max)
+        else:
+            range_string = ''.join([str(unichr(l)) for l in label['RANGE'][:label['NCRANG']]])
+        label.update({'RANGE': range_string})
+        # Process real coords:
+        x0 = struct.unpack('<f', ''.join([chr(x) for x in label.pop('X0V0')]))[0]
+        dx = struct.unpack('<f', ''.join([chr(x) for x in label.pop('DXV1')]))[0]
+        y0 = struct.unpack('<f', ''.join([chr(x) for x in label.pop('Y0V2')]))[0]
+        dy = struct.unpack('<f', ''.join([chr(x) for x in label.pop('DYV3')]))[0]
+        z0 = struct.unpack('<f', ''.join([chr(x) for x in label.pop('Z0V4')]))[0]
+        dz = struct.unpack('<f', ''.join([chr(x) for x in label.pop('DZV5')]))[0]

```

Comment:
Good idea, I will do the same with the opposite direction (`struct.pack`), because it's also repeated several times.
Where would you put these small functions? Define them inside the `_read_label`-function or directly on the module level?